sql/core.go

```diff
@@ -259,6 +259,10 @@ type StatisticsTable interface {
 	Table
 	// NumRows returns the unfiltered count of rows contained in the table
 	NumRows(*Context) (uint64, error)
+	// NextAutoIncrementValue returns the next autoincrement value for a table and 0 if autoincrement does not apply.
+	NextAutoIncrementValue(ctx *Context) (int64, error)
```

Comment:
This should be on AutoIncrementTable (or use the existing interface, should work fine)